sycl/include/CL/sycl/detail/device_impl.hpp

```diff
@@ -112,7 +112,7 @@ class device_impl_pi : public device_impl {
   ~device_impl_pi() {
     if (!m_isRootDevice) {
       // TODO catch an exception and put it to list of asynchronous exceptions
-      CHECK_OCL_CODE_NO_EXC(RT::piDeviceRelease(m_device));
+      CHECK_OCL_CODE_NO_EXC("device_impl.hpp", RT::piDeviceRelease(m_device));
```

Comment:
This seems brittle if some day the file is renamed...
Why no using `__FILE__` everywhere?
You can probably just keep the file name from the full path when you display it if you think it is too verbose.

Reply:
The objections to using  __FILE__ , in addition to displaying full path, is that emitting the filename may expose IP too, especially when used in header files. We may avoid this by explicitly requiring the user to supply a name explicitly.

Reply:
Unfortunately this implementation cannot use very modern C++, so you cannot have a `constexpr` function just cleaning up at compile-time the string returned by `__FILE__`... :-(
That looks like yet-another-good-motivation for modern C++! :-)

Reply:
I have an implementation to do so in c++11 that I've shared with Sindhu.

Reply:
Tried the following:
```
    1 #include <cstdio>
    2
    3 constexpr const char *GetRelFileName(const char *FullName,
    4                                      const char *FileName) {
    5   return *FullName == '\0'
    6              ? FileName
    7              : (*FullName == '/' ? GetRelFileName(FullName + 1, FullName + 1)
    8                                  : GetRelFileName(FullName + 1, FileName));
    9 }
   10
   11 int main() {
   12   const char *Res = GetRelFileName(__FILE__, __FILE__);
   13   printf(" %s \n", Res);
   14 }
```
It really prints filename without path. Unfortunately the final binary still contains full name(with path).
@erichkeane Do you have the same problem in your implementation?

Reply:
No, my version does not emit the full path to the binary.

Reply:
@erichkeane  @schittir Any news on c++ 11 implementation of stripping path from _FILE_ ?

Reply:
@keryell: The version I have given @schittir doesn't depend on opt level, none of the major 4 compilers emit anything but the file name.

@romanovvlad : Last I had heard is that @schittir is still trying to find a way to give a good interface for how to do this.